app/src/androidTest/java/com/github/se/project/ui/lesson/RequestedLessonsTest.kt

```diff
@@ -245,5 +239,30 @@ class LessonsRequestedScreenTest {
     composeTestRule.onNodeWithText("Instant Tutoring").performClick()
     verify(navigationActions).navigateTo(anyString())
   }
+
+  @Test
+  fun testFilterDialogInteraction() {
+    // Setup the initial content for RequestedLessonsScreen
+    composeTestRule.setContent {
+      RequestedLessonsScreen(
+          listProfilesViewModel, lessonViewModel, mockNetworkStatusViewModel, navigationActions)
+    }
+
+    // Step 1: Click on the "Filter by subjects" button (assumed to be in the UI)
+    // This will trigger the display of the FilterDialog
+    composeTestRule.onNodeWithTag("filterButton").performClick()
+
+    // Step 2: Verify that the FilterDialog opens and displays the subjects (PHYSICS and ANALYSIS)
+    composeTestRule.onNodeWithText("Physics").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Math Tutoring").assertIsDisplayed()
+
+    // Step 3: Simulate user selecting a subject (e.g., selecting Physics)
+    composeTestRule.onNodeWithText("Physics").performClick()
+
+    // Step 5: Simulate user dismissing the FilterDialog
+    // composeTestRule.onNodeWithContentDescription("Dismiss").performClick()
+
```

Comment:
Not sure to understand the comments here! Does some code miss or should you remove the comments?!? (for step 5 and 6)

Reply:
Thanks I am removing them right now